### PR TITLE
Added type overrides in config file.

### PR DESCRIPTION
**Pull request by roflolilolmao**
Whilst [adding support for `pgtype`][pr15], I realized that there was no perfect way to set up hard coded types. Allowing users to override types should allow them to have total control over the generated code.

These changes should make the `--int32-type` and `--uint32-type` arguments obsolete for users using the config file. Being completely optional and as this is an advanced use of `gendal` which should be properly documented in `README.md`, these changes are not documented in `gendal.toml`.

To override default behaviour of types generation, add an override by adding sections of the form:
```toml
[[TypeOverrides]]
DatabaseType = "database type name"
Type_ = "go_type"
NilValue = "go_nil_value"
NullableType = "nullable_go_type"
NullableNilValue = "nullable_go_nil_value"
```

If a section is incomplete, i.e. it does not contain the four values defining an override, a warning will be output to the std error stream.

[pr15]: https://github.com/turnkey-commerce/gendal/pull/15

### CODE REVIEW

**Comment by turnkey-commerce** [comment]
@roflolilolmao, thanks looks like a nice addition will review it shortly.

File: README.md [hunk 1]
```diff
@@ -456,6 +481,72 @@ for _, user := range users {
...
+```toml
+[[TypeOverrides]]
+DatabaseType = "SMALLINT"
+Type_ = "int16"
+NilValue = "0"
+NullableType = "*int16"
+NullableNilValue = "nil"
+```
+
+To us a go `string` and `*string` (with `NULL` values) for PostgreSQL's `TEXT`,
```
Comment (turnkey-commerce) [inline README.md:511]: Minor typo: us => use

**Review by turnkey-commerce — COMMENTED** [review]
Minor typo in README, otherwise looks good!

File: README.md (hunk 1 above)
Comment (roflolilolmao) [reply README.md:511]: Good catch, thanks!

**Comment by turnkey-commerce** [comment]
I'll go ahead and merge and fix that minor typo separately.

**Comment by roflolilolmao** [comment]
I probably forgot to `git add` my fix. Oops… Thanks!

**Comment by turnkey-commerce** [comment]
No problem, it was a small change :)

**Pull request merged**